Modix.Services.Test/Core/DiscordSocketListeningBehaviorTests.cs

```diff
@@ -0,0 +1,567 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+using Moq;
+using Moq.AutoMock;
+using NUnit.Framework;
+using Shouldly;
+
+using Discord;
+using Discord.Rest;
+using Discord.WebSocket;
+
+using Modix.Common.Messaging;
+using Modix.Services.Core;
+
+using Modix.Common.Test.Mocks;
+
+namespace Modix.Services.Test.Core
+{
+    [TestFixture]
+    public class DiscordSocketListeningBehaviorTests
+    {
+        #region Fakes
+
+        #pragma warning disable CS0067 // Unused members (no shit, that's how interfaces work)
+        private class FakeDiscordSocketClient : IDiscordSocketClient
```

Comment:
Do we actually need this entire half-implemented class? I'm no expert, but I thought you could mock an interface with Moq and just set implementations for specific methods that you need. https://github.com/Moq/moq4/wiki/Quickstart#methods

Reply:
Yeah, that's the benefit of using Moq. Unfortunately we can't use Moq for this one specifically, for 2 reasons.

1. Moq does not record invocations of event `add` and `remove` methods. So, unlike with other mocking frameworks, like `NSubstitute` you can't say "did the UUT subscribe to this event?". This isn't that big of a deal, since we can observe changes in functionality when a subscription does and does not occur, however...
2. Moq does not support non-standard event types. Such as all of the Discord events, which return a `Task` that needs to be `await`ed (or really, a set of `Task`s, one from each handler). So, if we used a `Mock<IDiscordSocketClient>` there's literally no way to get it to raise its events. You know, the things we're testing here.

Reply:
Ah, that makes sense